src/Compilers/CSharp/Portable/Lowering/BoundTreeToDifferentEnclosingContextRewriter.cs

```diff
@@ -131,11 +131,27 @@ public override BoundNode VisitAwaitableInfo(BoundAwaitableInfo node)
             var getAwaiter = (BoundExpression?)this.Visit(node.GetAwaiter);
             var isCompleted = VisitPropertySymbol(node.IsCompleted);
             var getResult = VisitMethodSymbol(node.GetResult);
-            var runtimeAsyncAwaitMethod = VisitMethodSymbol(node.RuntimeAsyncAwaitMethod);
 
             _placeholderMap.Remove(awaitablePlaceholder);
 
-            return node.Update(rewrittenPlaceholder, node.IsDynamic, getAwaiter, isCompleted, getResult, runtimeAsyncAwaitMethod);
+            BoundCall? runtimeAsyncAwaitCall = null;
+            var runtimeAsyncAwaitCallPlaceholder = node.RuntimeAsyncAwaitCallPlaceholder;
+            var rewrittenRuntimeAsyncAwaitCallPlaceholder = runtimeAsyncAwaitCallPlaceholder;
+            if (rewrittenRuntimeAsyncAwaitCallPlaceholder is not null)
+            {
+                rewrittenRuntimeAsyncAwaitCallPlaceholder = runtimeAsyncAwaitCallPlaceholder!.Update(VisitType(rewrittenRuntimeAsyncAwaitCallPlaceholder.Type));
+                _placeholderMap.Add(runtimeAsyncAwaitCallPlaceholder, rewrittenRuntimeAsyncAwaitCallPlaceholder);
+                runtimeAsyncAwaitCall = (BoundCall?)this.Visit(node.RuntimeAsyncAwaitCall);
+                _placeholderMap.Remove(runtimeAsyncAwaitCallPlaceholder);
+            }
+#if DEBUG
```

Comment:
Do we need this directive?

[Test](https://lab.razor.fyi/#jVJLbhNBEBWYBfQKcYLWsJlBSV_ACpIZmwByiEUSIoGQaI9rxqXUVJuuHhsH-TIs2HEV1mw4AjdggWZsbMeQiN501atXr15_1I-WUgPvCm9Lk8mDb61KkAt9MpcAZVttZyZ1RJAFdCzmEBg8ZjcwnpdlFeyQYIfTRVuwk4CZXF8xqRtBhy3NBXdpfeQPO9Dp2IMdIRfX4ebUyoW0lcrIiujVidUnpbXWEmzATE8djvSRRY6TBl4W65U6Fkdgzj0G6CNDHD0DImeipL0mDZ0jPdQHOvgKNjDmOh4uBRtRvVjHDzHX3d6Ts8M1AiSwoa6jenVhWBWmIwI-xLklga3ZW5rAI8ybdKEW6tHXVuo4x8KkJ2PrJwPrBY4nzQvFbrnrg8d6FTZ95hzDuG-5qGwBr8ELOo53cjPwMEWYJZuOp2BD5UHitwyzOKoE9r0TmvN-cBfAeAk-2tNRfTtR8i5RSVupK95SV06QbG3kfxx2iNzsjMXmENeiW1ZeVkT1z0sdB_gY_qj9GzY9rsGt9rpQ4mXjpA9ToPgvxLwCAitw5Ri9EtezbnJe844g2JEN9phpvnrOWurFbaQ39-5-__L55y-4f-f9rd8) shows that the branch would be pruned in RELEASE mode